gr-utils/python/blocktool/core/parseheader.py

```diff
@@ -0,0 +1,278 @@
+        self.target_file = file_path
+        self.initialize()
+        self.validate()
+
+    def initialize(self):
+        """
+        initialize all the required API variables
+        """
+        self.module = self.target_file
+        for dirs in self.module:
+            if not os.path.basename(self.module).startswith(Constants.GR):
+                self.module = os.path.abspath(
+                    os.path.join(self.module, os.pardir))
+        self.modname = os.path.basename(self.module)
+        self.filename = os.path.basename(self.target_file)
+        self.targetdir = os.path.dirname(self.target_file)
+        for dirs in os.scandir(self.module):
+            if dirs.is_dir():
+                if dirs.path.endswith('lib'):
+                    self.impldir = dirs.path
+        self.impl_file = os.path.join(self.impldir,
+                                      self.filename.split('.')[0]+'_impl.cc')
+
+    def validate(self):
+        """ Override the Blocktool validate function """
+        BlockTool._validate(self)
+        if not self.filename.endswith('.h'):
+            raise BlockToolException(
+                'Cannot parse a non-header file')
+
+    def get_header_info(self):
+        """
+        PyGCCXML header code parser
+        magic happens here!
+        # : returns the parsed header data in python dict
+        # : return dict keys: namespace, class, io_signature, make,
+        #                properties, methods
+        # : Can be used as an CLI command or an extenal API
+        """
+        gr = self.modname.split('-')[0]
+        module = self.modname.split('-')[-1]
+        generator_path, generator_name = utils.find_xml_generator()
+        xml_generator_config = parser.xml_generator_configuration_t(
+            xml_generator_path=generator_path,
+            xml_generator=generator_name,
+            compiler='gcc')
+        decls = parser.parse(
+            [self.target_file], xml_generator_config)
+        global_namespace = declarations.get_global_namespace(decls)
+
+        # namespace
+        try:
+            self.parsed_data['namespace'] = []
+            ns = global_namespace.namespace(gr)
+            if ns is None:
+                raise BlockToolException
+            main_namespace = ns.namespace(module)
+            if main_namespace is None:
+                raise BlockToolException('namespace cannot be none')
+            self.parsed_data['namespace'] = [gr, module]
+            if main_namespace.declarations:
+                for _namespace in main_namespace.declarations:
+                    if isinstance(_namespace, declarations.namespace_t):
+                        if Constants.KERNEL not in str(_namespace):
+                            main_namespace = _namespace
+                            self.parsed_data['namespace'].append(
+                                str(_namespace).split('::')[-1].split(' ')[0])
+        except RuntimeError:
+            raise BlockToolException(
```

Comment:
Why re-raising and not just handling the `RuntimeError` as is? As far as I understand, the `BlockToolException` is not specialized but just a wrapper for the general `exception`